### PR TITLE
Fix `Route` serialization round-trip

### PR DESCRIPTION
**Pull request by TheBlueMatt**
When the `max_total_routing_fee_msat` parameter was added to `RouteParameters`, the serialization used `map` to get the max fee, accidentally writing an `Option<Option<u64>>`, but then read it as an `Option<u64>`. Thus, any `Route`s with a `route_params` written will fail to be read back.

Luckily, this is an incredibly rarely-used bit of code, so only one user managed to hit it.

### CODE REVIEW

**Comment by jkczyz** [comment]
> There is also a minor test failure due to a BOLT12 test expecting to find an empty route, which we however refuse to read. Expecting to find the empty route might also be a mistake here? (cc @jkczyz)

`fails_paying_for_bolt12_invoice` isn't exhaustively testing the possible failures, but was meant to test anything after receiving an `Ok` result from the router whereas `fails_finding_route_for_bolt12_invoice` hits the `Err` case. We can probably just drop `fails_paying_for_bolt12_invoice` and test those error cases directly via `send_payment_with_route`. I'm not sure how well the coverage is there.

File: lightning/src/util/test_utils.rs [hunk 1]
```diff
@@ -143,6 +143,7 @@ impl<'a> Router for TestRouter<'a> {
 		&self, payer: &PublicKey, params: &RouteParameters, first_hops: Option<&[&ChannelDetails]>,
 		inflight_htlcs: InFlightHtlcs
 	) -> Result<Route, msgs::LightningError> {
+		let route_res;
 		if let Some((find_route_query, find_route_res)) = self.next_routes.lock().unwrap().pop_front() {
```
Comment (jkczyz) [inline lightning/src/util/test_utils.rs:147]: Maybe to save rustfmt from inevitably mangling the existing line, re-write as:

```rust
let next_route = self.next_routes.lock().unwrap().pop_front();
let route_res = if let Some((find_route_query, find_route_res)) = next_route {

**Review by jkczyz — APPROVED** [review]
